trieste/models/gpflow/models.py

```diff
@@ -19,19 +19,21 @@
 import gpflow
 import tensorflow as tf
 import tensorflow_probability as tfp
+from gpflow.conditionals.util import sample_mvn
 from gpflow.models import GPR, SGPR, SVGP, VGP
 from gpflow.utilities import multiple_assign, read_values
+from gpflow.utilities.ops import leading_transpose
 
 from ...data import Dataset
 from ...types import TensorType
 from ...utils import DEFAULTS, jit
-from ..interfaces import TrainableProbabilisticModel
+from ..interfaces import FastUpdateModel, TrainableProbabilisticModel
 from ..optimizer import BatchOptimizer, Optimizer
 from .interface import GPflowPredictor
 from .utils import assert_data_is_compatible, randomize_hyperparameters, squeeze_hyperparameters
 
 
-class GaussianProcessRegression(GPflowPredictor, TrainableProbabilisticModel):
+class GaussianProcessRegression(GPflowPredictor, TrainableProbabilisticModel, FastUpdateModel):
```

Comment:
since now we have this new interface, we could instead have a subclass, say `FastGaussianProcessRegression` that inherits both `GaussianProcessRegression` and `FastUpdateModel`, wouldn't this be a bit cleaner? @apaleyes any thoughts?

Reply:
should this be another PR?

Reply:
many thoughts, but i guess we agreed to implement it like that to unblock @vpicheny